### ISSUE TITLE
For users: Report wrong IP geo location here

### ISSUE THREAD

**Issue by jimaek** [jsdelivr/globalping#268]
If you are hosting a probe and know that the resolved location is wrong:

1) If the city is wrong but the country is correct, you can set the correct city yourself after adopting the probe on our [dashboard](https://dash.globalping.io/).

2) If the country is wrong, please report it here. We will use this data to improve our algorithm. Please do it only if you're fine with making your IP public.

   Please comment with:

   ```
   Real location:
   Resolved location:
   IP address:
   ```

**Comment by yoursunny** [comment]
Real location: Dallas TX, USA
Resolved location: "(New Zion, US, NA) (lat: 33.7957 long: -80.0148)"
IP address: 108.174.203.*

This is LigaHosting VPS, described to be in Dallas, confirmed via ping latency.

**Comment by gshapiro** [comment]
Real location: Milpitas, US, NA
Resolved location: Santa Clara, US, NA
IP address: 98.51.131.164

It was close (within 8 miles), not sure what fidelity you are looking for.  The probe reported:

[2023-01-13 22:32:50] [INFO] [695] [api:connect] connected from (Santa Clara, US, NA) (lat: 37.3541 long: -121.9552)

**Comment by Ne00n** [comment]
Real Location: FFM
Resolved Location: BE
Probe: Berlin, DE, EU, 210993

**Comment by skapytek** [comment]
Real Location: Wolfsegg am Hausruck, Upper Austria, Austria
Resolved Location: Laakirchen
Probe: Laakirchen, AT, EU, 8412 -- T-Mobile Austria GmbH
If its not a problem i wont write my IP publicly cause ive installed the probe at my privat home.

**Comment by skapytek** [comment]
Real Location: Wolfsegg am Hausruck, Upper Austria, Austria
Resolved Location: Laakirchen
Probe: Laakirchen, AT, EU, 8412 -- T-Mobile Austria GmbH
IP:  84.113.52.3

**Comment by jimaek** [comment]
We released a new version of our geo IP resolution logic. I invite everyone to check their probes and provide us with feedback.

**Comment by gshapiro** [comment]
> We released a new version of our geo IP resolution logic. I invite everyone to check their probes and provide us with feedback.

@jimaek What sort of accuracy are you interested in having reported?  My probe geolocation is within 3 miles of actual (though different city).

**Comment by jimaek** [comment]
3 miles is considered good for us :) Once we release our dashboard you will be able to correct it manually too

**Comment by gshapiro** [comment]
Real location: Fremont, US, NA (37.4719,-121.9204)
Resolved location:  Astoria, US, NA (46.1558,-123.7978)
IP address: 65.19.141.130

(about 750 miles difference)

**Comment by jimaek** [comment]
Thanks for reporting, it should start showing the correct location very soon, within the next few days once our caches expire and your probe does a scheduled restart.

**Comment by alissonlauffer** [comment]
Real location: Fortaleza, CE, Brazil (-3.7139,-38.5409)
Resolved location: Lisbon, Portugal (38.7167,-9.1333)
IP address: 185.148.113.186

The probe's geoip seem to be floating between Fortaleza and Lisbon, I'm not sure why though. Probably the company doesn't split the Fortaleza and Lisbon ranges correctly (i.e. `185.148.113.5` is in the same /24 range, but it's in Lisbon).

**Comment by jimaek** [comment]
Interesting, ipinfo and ip2location show Portugal, only maxmind reports Brazil. I'll look into it, thanks for reporting

**Comment by alissonlauffer** [comment]
@jimaek hello, any update regarding it? Thanks!

**Comment by risturiz** [comment]
Real location: Cúcuta, Norte de Santander, Colombia ( maxmind.com )
Resolved location: Barranquilla ( ipinfo.io ) & Pamplona ( ip2location.com )
IP address: 179.32.121.124

**Comment by ellieschieder** [comment]
Many German nodes report the wrong geolocation. NO node that is hosted by Hetzner should ever get assigned any other location then either Nuremberg or Falkenstein for now. This are there only possible geolocations. Also it’s looks a bit excessive when nearly 50% of my requests get assigned to probes in the same ASN. Even more when they are all hosted by just one provider.

**Comment by jimaek** [comment]
Hey, thanks for the feedback

1. Geolocation is quite complicated, we want to avoid manual rules per provider and have a system that works for everyone. Based on what I see the majority of Hetzner German locations are correctly assigned to those 2 cities. But there are a few exceptions, I'll try to find out why
2. We're currently working on launching our dashboard. It will allow all probe owners to manage them and fix the location of the probes. I hope this will be enough in most cases
3. "Also it’s looks a bit excessive when nearly 50% of my requests get assigned to probes in the same ASN. " can you give a few examples? Do you mean when you run "from Germany" you get Hetzner too often?

**Comment by arraisfilho** [comment]
Real location: Araguaína, Tocantins, BR
Resolved location: Brasília, Federal District, BR
IP address: 200.213.107.174

**Comment by LindenCX** [comment]
Real location: Barcelona, Spain
Resolved location: Scranton, US, NA
IP address: 194.26.100.1

**Comment by jimaek** [comment]
@LindenCX hey is your probe online and connected? Can you leave it up while we check the issue?

**Comment by LindenCX** [comment]
@jimaek Not my probe sorry, was just reporting one I commonly see in the US that has high latency to basically everything since it's not in the US.

There's a few more if you want me to drop them here, or I can toss you an email if you prefer with the list.

**Comment by jimaek** [comment]
Yes please email me dakulovgr@gmail.com the IPs and more details about how you made the list. Thanks!

**Comment by su-haris** [comment]
Real location: Bucharest, Romania
Resolved location: Belgium
IP address: 94.177.106.61

**Comment by jimaek** [comment]
@su-haris is the probe connected right now? Can you show the log with the location? We're not sure why it would show Belgium

**Comment by su-haris** [comment]
@jimaek the probe is indeed online, but surprisingly, it does not show an IPv4 address on the dashboard.
This is the IPv6 address: 2a0f:9400:7e11:876::1

**Comment by MartinKolarik** [comment]
@su-haris thanks, that particular IP is indeed reported to be in 5 different countries, I'll check what we can do here.

**Comment by jimaek** [comment]
For everyone having issues with correct location detection we now have a new flow:
1. Please adopt your probe and manual change the location in the dashboard.
2. If the location dropdown does not have the correct data then please visit this new tool https://geodebug.globalping.dev/
3. At the end there is a location correct form you can submit!
